src/link/languages.js

```diff
@@ -340,5 +342,85 @@ const languagesWithWikis = [
 	'mus',
 ];
 
-export const isLanguageWithWiki = ( language ) =>
+const limit = defaultLanguages.length;
+const languages = getLanguages();
+
+const isLanguageWithWiki = ( language ) =>
 	languagesWithWikis.indexOf( language ) !== -1;
+
+const normalize = ( result, language ) => {
+	if ( isLanguageWithWiki( language ) ) {
+		result.push( {
+			name: languages[ language ][ 2 ],
+			code: language,
+		} );
+	} else if ( language === 'en-simple' ) {
+		result.push( {
+			name: languages[ language ][ 2 ],
+			code: 'simple',
+		} );
+	}
+	return result;
+};
+
+const getLocalized = ( language, userLang ) => {
+	const localizedName = new Intl.DisplayNames( [ userLang ], {
+		type: 'language',
+	} );
+	const invalid = [
+		'bat-smg',
+		'be-x-old',
+		'cbk-zam',
+		'fiu-vro',
+		'map-bms',
+		'roa-rup',
+		'zh-classical',
+		'zh-min-nan',
+		'zh-yue',
+		'zh-cdo',
+	];
+
+	if ( invalid.indexOf( language ) === -1 ) {
+		return localizedName.of( language ).toLowerCase();
+	}
+	return false;
+};
+
+export const filterLanguages = ( target, lang, setValue, setItems ) => {
+	setValue( target );
+	const targetLang = target.toLowerCase().trim();
+
+	if ( targetLang === '' ) {
+		setItems( defaultFilter() );
+		return;
+	}
+
+	const filtered = Object.keys( languages )
+		.filter( ( language ) => {
+			const localized =
+				Intl.DisplayNames && getLocalized( language, lang );
```

Comment:
I would suggest keeping the check as close as possible to where Intl.DisplayNames is being used so it is easier to keep track that those 2 needs are needed together.